crates/rmf_site_editor/src/mapf_rse/negotiation/visual.rs

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2024 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use super::*;
+use crate::site::line_transform;
+use bevy::ecs::hierarchy::ChildOf;
+use bevy::math::prelude::Rectangle;
+
+pub const DEFAULT_PATH_WIDTH: f32 = 0.2;
+
+#[derive(Component)]
+pub struct PathVisualMarker;
+
+pub fn visualise_selected_node(
+    mut commands: Commands,
+    negotiation_task: Query<&NegotiationTask>,
+    debug_data: ResMut<NegotiationDebugData>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    path_visuals: Query<Entity, With<PathVisualMarker>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    robots: Query<&Affiliation<Entity>, With<Robot>>,
+    robot_descriptions: Query<&CircleCollision, (With<ModelMarker>, With<Group>)>,
+    current_level: Res<CurrentLevel>,
+) {
+    // Return unless complete
+    let Ok(negotiation_task) = negotiation_task.single() else {
+        return;
+    };
```

Comment:
There is a slightly odd mix of resources (i.e. `ResMut<NegotiationDebugData>`) and queries that we only expect to match a single entity (`Query<&NegotiationTask>`).
Imho a query that matches a single entity is a bit of a footgun, if something is expected to be an app wide single instance it should be a resource. If we expect it to be at most one per site then it should be an entity that is a child of the site.